app/serializers/activitypub/undo_emoji_reaction_serializer.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class ActivityPub::UndoEmojiReactionSerializer < ActivityPub::Serializer
+  attributes :id, :type, :actor
+
+  has_one :object, serializer: ActivityPub::EmojiReactionSerializer
+
+  def id
+    [ActivityPub::TagManager.instance.uri_for(object.account), '#emoji_reactions/', object.id, '/undo'].join
```

Comment:
Related to the deletion issue, I think this payload should also include the domain, so there is no ambiguity regarding emojis with the same shortcode, but different domains.